### ISSUE TITLE
would tomboy-ng support ibus input method？

### ISSUE THREAD

**Issue by lennjf** [tomboy-notes/tomboy-ng#311]
thank developers very much, I like this app.
I am using tomboy-ng， if it support ibus, it would be perfect

**Comment by davidbannon** [comment]
Thanks for the feedback lennjf. I don't use ibus myself, in fact, I don't fully understand what it does.
I do understand its an operating system thing, not an application thing. So, if you have ibus installed and configured, it should work with tomboy-ng. Or, so I understand.

You have not mentioned which operating system or version of tomboy-ng you are using, so, assuming Linux and Qt5 (??).

Please type -
`$>  echo "$QT_IM_MODULE"`

That should return "ibus", if not, then you need to look at your configuration. But I may well be wrong about that.

Davo

**Comment by lennjf** [comment]
thank you for reply this issue, Mr. Davo.

yes, when I run "echo "$QT_IM_MODULE" , I got "ibus".

I have both QT5 and QT6 installed on my arch linux system,
my tomboy-ng version is : 0.39-1

ibus is a input method engine for different languages input, as qt programs work it with "libibusplatforminputcontextplugin.so", but I don't know anything of pascal.

I think many users of different language, not Latin language， need it to type words when using tomboy-ng

**Comment by davidbannon** [comment]
Hi lennjf, thanks for getting back to me. I think we, together, can get to to bottom of this problem. But you will need to help me !
FreePascal apps are usually 100% UTF8 compliant and I think what happens is the Qt engine watches you type in an ibus sequence, it looks up the corresponding utf8 character and sends it to my code. But there is a lot more going on ...

Firstly, its possible one of the components I use, KMemo, cannot handle it. It cannot, for example, handle right-to-left text. So, as a test, can you please open the tomboy-ng main search window and see if you can enter the extended character into the search field. That will not go via KMemo and will tell us if its a kmemo issue. Can you see the character there ?

Next, can you teach me how to use ibus ? It seems incredibly badly documented. So, what exactly do you do to use it ?  What characterset are you hoping to display ?  I'd like to make a small test app that we can use ...

tomboy-ng does display all the utf8 set that I have tested, the Insert Char tool lets you set it up to insert a single char if you know its UTF8 code, so, have you tried that ?  As a test, I know its not suitable for free typing !

Davo

**Comment by lennjf** [comment]
hi, Davo, I want to say, for me, it's okay to use tomboy-ng without ibus. 
actually， I don't know how to use ibus for development， when someone want to type Arabic、Chinese、Japanese...etc, on linux, they would use ibus.
it's hard to understand it for non-English users, thank you for your time,  let's give it up, have a good day

**Comment by lennjf** [comment]
something to say,  before I install this app with the "pkg.tar.zst" release package, because I use ARCH linux.

but now I download the "amd64Qt6.deb" release package, and use "debtab" to transform it to  "pkg.tar.zst", then install it, it works, now I can use ibus inputmethod,

thanks Davo again

**Comment by davidbannon** [comment]
OK, now, that makes sense !   I only make a gtk2 version for Pacman, not a lot of users and its simple. It sounds like gtk2 does not work with iBus but Qt6 (and probably but not certain) Qt5 does work with iBus.

Now, that probably means i need to make a Qt6 pacman too. 

Question ?  To get the Qt6 version working, did you get the 'special' LibQt6Pas library from https://github.com/davidbannon/libqt6pas/releases or did the library in the Arch Repo work OK ?   I ask because, at present, neither Debian nor Ubuntu support Qt6 and I'd really rather use it, it works better than the Qt5 ones, mainly because of Wayland.

Davo

**Comment by lennjf** [comment]
I run "pacman -Q |grep pas" , it shows that I have "qt6pas 6.2.7-2", it should be installed from arch repo automatically when I install tomboy-ng, yes, it works OK, and my tomboy-ng installation package is from "debtab", a tool to transform deb to arch package.
but, I don't how to adjust the font size of the windows of tomboy,
the font on notes are good, but the font on windows, is too small, maybe it because I am using XFCE desktop, but in other QT app, the font is normal

![image](https://github.com/tomboy-notes/tomboy-ng/assets/41781491/49af2a98-8985-45ef-9403-77fb2d437a95)
![image](https://github.com/tomboy-notes/tomboy-ng/assets/41781491/0d0c8570-bd3e-4e78-a5f3-1c55c3807a68)

**Comment by davidbannon** [comment]
Hmm, qt6pas 6.2.7-2 is a little older than the current one, only a small difference, the newer one has one obsolete symbol removed that apparently upset some other Qt6 libraries. If it works for you, I guess that all we need to know !
I will have to look into debtab, the way I make the current (gtk2) pacman package is a real pain !

Thanks for you input here, very valuable !

Davo

**Comment by davidbannon** [comment]
Yes, I see what you mean. A lot of empty space there. I purposely leave quite a bit of space because users, with bad eyesight, use oversized fonts, x1.5 or even x2.  You seem to have the opposite problem !

I do not "set" the font size in tomboy-ng, good practice says take the Desktop "Application Font" and use it, then, should be the same for all apps. 

I don't know what Desktop you use. I use Mate, and go to the "Appearance" setting, "Application Font". Its set to Cantarell Regular 11 in my case and most apps use that setting.

So, be careful, if you change your Application Font, it should change how its displayed for all applications that just use that default font. 

If you have a (very) high resolution screen, high DPI, that can sometimes cause scaling problems. There will be settings that allow you to scale graphics at a different rate than text. But again, changing settings like that would change most applications, not just tomboy-ng.

I now have a Qt6 tomboy-ng package available for Arch. In future, that will be the only one I make for pacman, my tests agree with yours, its works well.

Davo

**Comment by lennjf** [comment]
even I adjust the font size in "Appearance-font", the font of  right click context menu of tomboy changed, but the font of windows not changed, I will get used to it,  it's not a problem

**Comment by davidbannon** [comment]
It sounds like lennjf now has what is needed. Thanks for the information lennjf, great to know that there is one more really good reason to use Qt !

Davo
